### PR TITLE
Avoid lowpass=0 in brainvision data

### LINKED ISSUE #10516: Some BrainVision data contain Lowpass=0
#### Describe the bug
BrainVision data acquired with V-Amp without any hardware filtering are stored in header file as:
```
Channels
--------
#     Name      Phys. Chn.    Resolution / Unit   Low Cutoff [s]   High Cutoff [Hz]   Notch [Hz]    Gradient         Offset
1     EOGR        1          0.0488281 µV             DC                0              Off
2     EOGL        2          0.0488281 µV             DC                0              Off
3     FP3         3          0.0488281 µV             DC                0              Off
4     Fz          4          0.0488281 µV             DC                0              Off
```
The current code in brainvision.py, the filtering section don't set the lowpass filter only in the case the value is NaN or Off
```python
if len(lowpass) == 0:
    pass
elif len(set(lowpass)) == 1:
    if lowpass[0] in ('NaN', 'Off'):
        pass  # Placeholder for future use. Lowpass set in _empty_info
    else:
        info['lowpass'] = float(lowpass[0])
        if lp_s:
            # filter time constant t [secs] to Hz conversion: 1/2*pi*t
            info['lowpass'] = 1. / (2 * np.pi * info['lowpass'])

else:
    heterogeneous_lp_filter = True
    if lp_s:
        # We convert channels with disabled filters to having
        # infinitely relaxed / no filters
        lowpass = [float(filt) if filt not in ('NaN', 'Off')
                   else 0.0 for filt in lowpass]
        info['lowpass'] = np.min(np.array(lowpass, dtype=np.float64))
        try:
            # filter time constant t [secs] to Hz conversion: 1/2*pi*t
            info['lowpass'] = 1. / (2 * np.pi * info['lowpass'])

        except ZeroDivisionError:
            if len(set(lowpass)) == 1:
                # No lowpass actually set for the weakest setting
                # so we set lowpass to the Nyquist frequency
                info['lowpass'] = info['sfreq'] / 2.
                # not actually heterogeneous in effect
                # ... just heterogeneously disabled
                heterogeneous_lp_filter = False
            else:
                # no lowpass filter is the weakest filter,
                # but it wasn't the only filter
                pass
    else:
        # We convert channels with disabled filters to having
        # infinitely relaxed / no filters
        lowpass = [float(filt) if filt not in ('NaN', 'Off')
                   else np.Inf for filt in lowpass]
        info['lowpass'] = np.max(np.array(lowpass, dtype=np.float64))

        if np.isinf(info['lowpass']):
            # No lowpass actually set for the weakest setting
            # so we set lowpass to the Nyquist frequency
            info['lowpass'] = info['sfreq'] / 2.
            if len(set(lowpass)) == 1:
                # not actually heterogeneous in effect
                # ... just heterogeneously disabled
                heterogeneous_lp_filter = False

    if heterogeneous_lp_filter:
        # this isn't clean FP, but then again, we only want to provide
        # the Nyquist hint when the lowpass filter was actually
        # calculated from dividing the sampling frequency by 2, so the
        # exact/direct comparison (instead of tolerance) makes sense
        if info['lowpass'] == info['sfreq'] / 2.0:
            nyquist = ', Nyquist limit'
        else:
            nyquist = ""
        warn('Channels contain different lowpass filters. '
             'Highest (weakest) filter setting (%0.2f Hz%s) '
             'will be stored.' % (info['lowpass'], nyquist))
```

This lead to an error when you try e.g. to plot the data.
```
  File "<decorator-gen-292>", line 12, in plot_raw
  File "c:\users\atonin\pycharmprojects\mne-fork\mne-python\mne\viz\raw.py", line 227, in plot_raw
    decim, picks_data = _handle_decim(info, decim, lowpass)
  File "c:\users\atonin\pycharmprojects\mne-fork\mne-python\mne\viz\utils.py", line 1754, in _handle_decim
    decim = max(int(info['sfreq'] / (lp * 3) + 1e-6), 1)
ZeroDivisionError: float division by zero
```

### PR DESCRIPTION
**Pull request by Lychfindel**
#### Reference issue
Example: Fixes #10516 .

#### What does this implement/fix?
If in the header high cutoff is 0, then the lowpass filter is considered as disabled.

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
no opinion here. @sappelhoff thoughts?

**Review by sappelhoff — COMMENTED** [review]
I have never used a V-Amp or analyzed data produced by it, but the proposed changes look reasonable to me. If @Lychfindel can vouch for the V-Amp data being organized this way, I think it's fine to go ahead.

@Lychfindel did you record the data yourself and made sure the `0` gets entered when you 
have a BrainVision Recorder setting to disable **software filters**?

Note that **hardware filters** are surely still in place, even if neither the data nor the recording interface (I assume BrainVision Recorder) tells you about that. That's just a necessity to prevent aliasing of the signal when digitizing to some sampling frequency.

I guess we need to distinguish 3 types of filters:

1. hardware --> this filters the data during / prior to digitization of the analogue signal
2. software 1 --> this is applied in the recording interface (e.g. BrainVision recorder) and directly applied to the recorded data, cannot be reversed
3. software 2 --> this is applied e.g., in MNE-Python to already recorded and saved data ... can be changed and reversed at will

Example for 2. --> the Brain Products "Brain AMP DC" **always** samples at 5kHz -- even when you set the sampling frequency to 1000Hz in the recording interface. "Under the hood", the recording interface just applies an additional software filter and downsampling process prior to saving the data. This has been confirmed to me by Brain Products in a personal Email. :-)

**Comment by Lychfindel** [comment]
Yes, I recorded the data myself even last week, and I always disable software filters. The schema in the hdr file is always the same:
```
Brain Vision V-Amp Data Header File Version 1.0
...
Channels
--------
#     Name      Phys. Chn.    Resolution / Unit   Low Cutoff [s]   High Cutoff [Hz]   Notch [Hz]    Gradient         Offset
1     Fz          1          0.0488281 µV             DC                0              Off
2     Cz          2          0.0488281 µV             DC                0              Off
3     C3          3          0.0488281 µV             DC                0              Off
4     C4          4          0.0488281 µV             DC                0              Off
5     T7          5          0.0488281 µV             DC                0              Off
6     T8          6          0.0488281 µV             DC                0              Off
7     Pz          7          0.0488281 µV             DC                0              Off
8     P3          8          0.0488281 µV             DC                0              Off
9     P4          9          0.0488281 µV             DC                0              Off
10    EOGL        10         0.0488281 µV             DC                0              Off
11    EOGR        11         0.0488281 µV             DC                0              Off
```

**Review by sappelhoff — COMMENTED** [review]
cool, then I'd say: please add a detailed changelog entry about this bugfix and we can merge. CI failures seem unrelated

If you wanna go the extra mile, you'd record a new datafile with software filters ON, and check how that looks in the data and whether we can read it in mne.

**Comment by Lychfindel** [comment]
In the V-Amp is not possible to set hardware filters, but I did three more test recordings: without software filters, with software filters, with software filters except in one channel. All of them are correctly read by mne and in the set with one channel not filtered I receive the expected warnings
```
<ipython-input-10-4e1c2c12eafd>:1: RuntimeWarning: Online software filter detected. Using software filter settings and ignoring hardware values
  raw = read_raw_brainvision(vhdr_path)
<ipython-input-10-4e1c2c12eafd>:1: RuntimeWarning: Channels contain different highpass filters. Lowest (weakest) filter setting (0.00 Hz) will be stored.
  raw = read_raw_brainvision(vhdr_path)
<ipython-input-10-4e1c2c12eafd>:1: RuntimeWarning: Channels contain different lowpass filters. Highest (weakest) filter setting (250.00 Hz, Nyquist limit) will be stored.
  raw = read_raw_brainvision(vhdr_path)
```


[V-AMP_test.zip](https://github.com/mne-tools/mne-python/files/8488216/V-AMP_test.zip)

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -37,6 +37,8 @@ Bugs
 
 - Retain epochs metadata when using :func:`mne.channels.combine_channels` (:gh:`10504` by `Clemens Brunner`_)
 
+- Fix bug in :meth:`mne.io.brainvision._get_vhdr_info` when BrainVision data are acquired with V-Amp and disabled lowpass filter is marked with value 0 (:gh:`10517` by `Alessandro Tonin`_)
```
Comment (sappelhoff) [inline doc/changes/latest.inc:40]:
```suggestion
- Fix bug in :func:`mne.io.read_raw_brainvision` when BrainVision data are acquired with the Brain Products "V-Amp" amplifier and disabled lowpass filter is marked with value ``0`` (:gh:`10517` by `Alessandro Tonin`_)
```

**Review by sappelhoff — APPROVED** [review]
one nitpick, other than that, +1 to merge

File: doc/changes/names.inc [hunk 2]
```diff
@@ -426,6 +426,8 @@
 
 .. _Senwen Deng: https://snwn.de
 
+.. _Alessandro Tonin: https://wysscenter.ch/team/alessandro-tonin
```
Comment (larsoner) [inline doc/changes/names.inc:429]: @Lychfindel I used the `:newcontrib:` above and realized we needed an entry in `names.inc`, feel free to push to change the URL if this is not the one you want!

**Comment by larsoner** [comment]
Thanks @Lychfindel !

File: doc/changes/names.inc (hunk 2 above)
Comment (Lychfindel) [reply doc/changes/names.inc:429]: Sure thanks a lot, then I will remove it from my other PR!

**Pull request merged**